src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -744,7 +748,31 @@ private bool TakeStartLine(SocketInput input)
                 RequestUri = requestUrlPath;
                 QueryString = queryString;
                 HttpVersion = httpVersion;
-                Path = RequestUri;
+
+                MemoryPoolIterator2 pathBaseEnd;
+                bool caseMatches;
+
+                if (!string.IsNullOrEmpty(_pathBase) &&
+                    requestUrlPath.Length >= _pathBase.Length &&
+                    pathBegin.StartsWithIgnoreCase(_pathBase, out pathBaseEnd, out caseMatches) &&
```

Comment:
if _pathBase is a fixed string/char array; isn't it excessive to create a MemoryPoolIterator2 out of it for the purpose of comparison? Could also create a _pathBaseLower at Frame initialisation time; rather than doing it on each request per char again and again?